crates/router/src/core/payouts/retry.rs

```diff
@@ -318,15 +324,21 @@ pub async fn modify_trackers(
 
 pub async fn config_should_call_gsm_payout(
     db: &dyn StorageInterface,
-    merchant_id: &String,
+    merchant_id: &common_utils::id_type::MerchantId,
     retry_type: PayoutRetryType,
 ) -> bool {
     let key = match retry_type {
         PayoutRetryType::SingleConnector => {
-            format!("should_call_gsm_single_connector_payout_{}", merchant_id)
+            format!(
+                "should_call_gsm_single_connector_payout_{}",
+                merchant_id.get_string_repr()
+            )
         }
         PayoutRetryType::MultiConnector => {
-            format!("should_call_gsm_multiple_connector_payout_{}", merchant_id)
+            format!(
```

Comment:
these function can move into impl based functions

Reply:
The issue here is that I would have to define two functions, only if i could also take the payout retry enum as input in the common utils. But this enum is defined in router.